### PR TITLE
Add section data to Student Profile for high school students

### PR DESCRIPTION
**Pull request by jhilde**
This PR adds sections to the overview area of the Student Profile implementing #1078.

**Who does it affect?** All high school students. Any student in a school type other than "HS" remains as is.

**What does it look like?**
capture

**What do I need to know?**
If an educator has permission to view a student's profile (via districtwide, schoolwide, homeroom, grade level, section, etc), they can see the student's current sections with current average grade. 

If an educator has access to a section in which the student is currently enrolled, they will see the section number as a link (goes to section view). Otherwise, the section number will be plain text.

### CODE REVIEW

**Comment by jhilde** [comment]
@alexsoble, @kevinrobinson let me know if you want to take a look. Otherwise, I'll deploy. Should be solid. Would be great to get into production by the end of the week.

Tested on IE11 ✅

File: app/assets/javascripts/helpers/profile_details_style.jsx [hunk 1]
```diff
@@ -19,6 +19,14 @@
       marginBottom: 10,
       width: '100%',
     },
+    roundedBox: {
+      border: '1px solid #ccc',
+      padding: 15,
+      marginTop: 10,
+      marginBottom: 10,
+      width: '100%',
+      borderRadius: '5px 5px 5px 5px',
```
Comment (kevinrobinson) [inline app/assets/javascripts/helpers/profile_details_style.jsx]: You can just do `borderRadius: 5` here if these are all the same.

File: app/assets/javascripts/student_profile/profile_details.jsx [hunk 2]
```diff
@@ -164,6 +170,38 @@ import _ from 'lodash';
       return _.sortBy(events, 'date').reverse();
     },
 
+    styleEducators: function(section, column) {
+      const educatorNames = section.educators.map(educator => {
```
Comment (kevinrobinson) [inline app/assets/javascripts/student_profile/profile_details.jsx]: You can do `_.map(section.educators, 'full_name')` if you like.

File: app/assets/javascripts/student_profile/profile_details.jsx [hunk 3]
```diff
@@ -197,6 +238,39 @@ import _ from 'lodash';
...
+    renderSectionDetails: function() {
+      const sections = this.props.sections;
+      
+      // If there are no sections, don't generate the section
+      if (!sections || sections.length == 0) return null;
+
+      // If this student is not a high school student, don't generate the section
+      if (this.props.student.school_type != 'HS') return null;
+      
+      const columns = [
```
Comment (kevinrobinson) [inline app/assets/javascripts/student_profile/profile_details.jsx]: This and the related methods in this component seem like they might split out nicely to a `SectionsTable` component or something like that, to keep it isolated and keep this component from getting huger :)

File: app/controllers/students_controller.rb [hunk 4]
```diff
@@ -37,6 +37,9 @@ def show
       educators_index: Educator.to_index,
       access: student.latest_access_results,
       iep_documents: student.iep_documents,
+      sections: student.sections.select('sections.*, student_section_assignments.grade_numeric')
```
Comment (kevinrobinson) [inline app/controllers/students_controller.rb]: This is a lot of complexity inline with serialization code, factor out into a method?

FWIW `educators_index` is sent to the browser so you can serialize just ids and join client side if that were simpler, no need to change here just FYI.  And TIL that `#select` doesn't just work as "filter a list" but also in the SQL sense to pluck fields. :)

File: app/models/educator.rb [hunk 5]
```diff
@@ -151,6 +151,10 @@ def allowed_sections
     end
   end
 
+  def allowed_sections_index
+    allowed_sections.map { |section| section.id }
```
Comment (kevinrobinson) [inline app/models/educator.rb]: Not sure the value of adding to the `Educator` model here?  You could inline `current_educator_allowed_sections: `educator.allowed_sections.map(&:id)` in the controller or a helper there.

File: spec/javascripts/student_profile/student_profile_page_spec.jsx [hunk 6]
```diff
@@ -95,8 +106,27 @@ describe('StudentProfilePage integration test', function() {
...
+      it('renders student with 1 section', function() {
+        const el = this.testEl;
+        helpers.renderStudentProfilePage(el, '10', [], 0, 1, 'HS');
+        expect(el).toContainText('1 section');
+      });
 
+      it('renders student with 5 sections', function() {
+        const el = this.testEl;
+        helpers.renderStudentProfilePage(el, '10', [], 0, 5, 'HS');
+        expect(el).toContainText('5 sections');
```
Comment (kevinrobinson) [inline spec/javascripts/student_profile/student_profile_page_spec.jsx]: ❤️ these simple tests!

**Comment by jhilde** [comment]
Thanks for the great code suggestions @kevinrobinson!

**Comment by jhilde** [comment]
Looking good :ship ing.

**Comment by jhilde** [comment]
Looks good in production on IE11.

Logged one bug #1108 to fix grade sorting. Not worth rolling back, no errors thrown, just weird sorting by grade.

**Pull request merged**